wpimath/src/main/java/edu/wpi/first/math/controller/PIDController.java

```diff
@@ -127,38 +127,62 @@ public void close() {
    * @param kp The proportional coefficient.
    * @param ki The integral coefficient.
    * @param kd The derivative coefficient.
+   * @throws IllegalArgumentException if kp &lt; 0
+   * @throws IllegalArgumentException if ki &lt; 0
+   * @throws IllegalArgumentException if kd &lt; 0
    */
   public void setPID(double kp, double ki, double kd) {
     m_kp = kp;
     m_ki = ki;
     m_kd = kd;
+    if (kp < 0.0) {
```

Comment:
We explicitly chose not to add these because crashing the user's code the middle of runtime is frowned upon. We could print a warning and clamp to zero instead.

In either case, this PR should be rebased against development since it's a breaking change.

Reply:
Warning and clamping to zero would match the changes I made to the cpp code. Would it be better to match the cpp behavior of warning and clamping to zero in the constructor as well -- getting rid of all the throws? 
As an engineering choice is it better to, as in the existing cpp constructor, clamp all 3 to zero if an attempt is made to set any one of the negative? Or, for setting them after construction, set only the one that is being made negative? I'll happily defer to whatever lessons have been learned from experience.

Reply:
We throw in Java constructors because that's "Java expected practice", as bad as that is for embedded systems.

Reply:
In that case my inclination is to simply leave everything as-is except the comment on setIntegratorRange. The proposed parameter checks don't necessarily make things overall better -- just "differently weird" (in my opinion).

Reply:
Since the only change now is to the documentation I did not rebase on development.